src/Bicep.LangServer/Deploy/DeploymentHelper.cs

```diff
@@ -79,8 +81,20 @@ public class DeploymentHelper
 
                 try
                 {
-                    var updatedParametersFileContents = DeploymentParametersHelper.GetUpdatedParametersFileContents(documentPath, parametersFileName, parametersFilePath, parametersFileUpdateOption, updatedDeploymentParameters);
-                    parameters = JsonElementFactory.CreateElement(updatedParametersFileContents);
+                    if(parametersFilePath.EndsWith(".bicepparam") && bicepparamJsonOutput is {})
```

Comment:
There if checks aren't super clean imo. I was wondering ways make this cleaner:
- move logic for compiling bicepparam file from handler to here (this would remove the `null` check for `bicepparamJsonOutput`)
- make the `paramsFileName` not nullable (although it only needed for json files when no json parameter file path is provided and we create/overwrite one)